datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -263,31 +243,45 @@ where
 
         let logger = logger.to_owned();
 
-        stream::unfold(from, move |start| {
+        stream::unfold((from, to - from), move |(start, step)| {
             if start > to {
                 return None;
             }
-            let end = if start < *LOG_STREAM_FAST_SCAN_END {
-                (start + 100_000 - 1).min(to).min(*LOG_STREAM_FAST_SCAN_END)
-            } else {
-                (start + 1_000 - 1).min(to)
-            };
-            let new_start = end + 1;
-
+            let end = start + step;
             debug!(logger, "Requesting logs for blocks [{}, {}]", start, end);
-
+            let logger = logger.clone();
             let log_filter = log_filter.clone();
             Some(
-                eth.logs_with_sigs(&logger, start, end, addresses.clone(), event_sigs.clone())
-                    .map(move |logs| {
-                        logs.into_iter()
-                            .filter(move |log| log_filter.matches(log))
-                            .collect::<Vec<Log>>()
-                    })
-                    .map(move |logs| (logs, new_start)),
+                eth_get_logs(
+                    eth.clone(),
+                    logger.clone(),
+                    start,
+                    end,
+                    addresses.clone(),
+                    event_sigs.clone(),
+                )
+                .map(move |logs| {
+                    logs.into_iter()
+                        .filter(move |log| log_filter.matches(log))
+                        .collect::<Vec<Log>>()
+                })
+                .then(move |res| match res {
+                    Err(e) => {
+                        let string_err = e.to_string();
+                        if string_err.contains(TOO_MANY_LOGS_FINGERPRINT) {
+                            let new_step = (step / 10).max(1);
+                            debug!(logger, "Reducing log step"; "new_step" => new_step);
```

Comment:
How about `"Reducing block range size to scan for events"` + `"new_size"`, since that's what the environment variable is called?